docs/user/guides/private-submodules.rst

```diff
@@ -8,9 +8,12 @@ How to use private Git submodules
 If you are using private Git repositories and they also contain private Git submodules,
 you need to follow a few special steps.
 
-Read the Docs uses SSH keys (with read only permissions) in order to clone private repositories.
-A SSH key is automatically generated and added to your main repository, but not to your submodules.
-In order to give Read the Docs access to clone your submodules you'll need to add the public SSH key to each repository of your submodules.
+Read the Docs uses SSH keys (with read only permissions) for GitLab and Bitbucket in order to clone private repositories,
+this key is added to your main repository, but not to your submodules.
+For GitHub we make use of a temporary token generated using our :ref:`GitHub App <reference/git-integration:GitHub App>`.
+
+When a project is created, a SSH key is automatically generated.
+You can use this SSH key to give Read the Docs access to clone your private submodules.
```

Comment:
I would finish taking about SSH keys and submodules first and then explain the GitHub case.

```suggestion
When adding a private GitLab and/or a Bitbucket project,
Read the Docs will generate a SSH key (with read only permissions) and add it to the repository to be able to clone. This SSH key is not added to the submodules of the repository.
In case you need to clone the private submodules, you can add this SSH key on those repositories as well.

When adding a GitHub project, Read the Docs make use of a temporary token generated using our :ref:`GitHub App <reference/git-integration:GitHub App>` instead of SSH keys.
```